src/freezing/freezing_compare.py

```diff
@@ -1,179 +1,177 @@
+
+
+def get_additions(
+    frozen_csv: pd.DataFrame,
+    updated_snapshot: pd.DataFrame,
+    ) -> pd.DataFrame:
+    """Get added records from the updated snapshot.
+
+    Get all records that are present in the updated snapshot but not the main
+
+    Args:
+        frozen_csv (pd.DataFrame): The staged and validated frozen data.
+        updated_snapshot
+
+    Returns:
+        additions_df (pd.DataFrame): The records that have been added.
+    """
+    key_cols = ["reference", "period", "instance"]
+
+    # To do a right anti-join, we need to do a full outer join first, then
+    # take only rows that were marked as right-only by the indicator. After
+    # that, there will be copies of every column in both, but for the
+    # right-only rows the columns from the left df will be null, so they're
+    # all dropped afterwards.
+    outer_join = pd.merge(
+        frozen_csv,
+        updated_snapshot,
+        on=key_cols,
+        how="outer",
+        suffixes=("_old", ""),
+        indicator=True,
+    )
+    additions_df = outer_join[(outer_join._merge == "right_only")].drop(
+        "_merge", axis=1
+    )
+    additions_df = additions_df[
+        additions_df.columns[~additions_df.columns.str.endswith("_old")]
+    ]
+
+    if additions_df.shape[0] > 0:
+        additions_df["accept_changes"] = False
+        return additions_df
+    else:
+        freezing_logger.info("No additions found.")
+        return None
+
+
+def output_freezing_files(
+    amendments_df: pd.DataFrame,
+    additions_df: pd.DataFrame,
+    config: dict,
+    write_csv: Callable,
+    run_id: int,
+    ) -> bool:
+    """Save CSVs of amendments and additions for user approval.
+
+    Args:
+        amendments_df (pd.DataFrame): The records that have changed.
+        additions_df (pd.DataFrame): The records that have been added.
+        config (dict): The pipeline configuration
+        write_csv (callable): Function to write to a csv file. This will be the
+            hdfs or network version depending on settings.
+        run_id (int): The run id for this run.
+
+    Returns:
+        bool: True if the files were written successfully.
+    """
+
+    freezing_changes_to_review_path = config["freezing_paths"]["freezing_changes_to_review_path"]
+    FreezingLogger.info("Outputting changes to review file(s).")
+    tdate = datetime.now().strftime("%y-%m-%d")
+    survey_year = config["years"]["survey_year"]
+
+    # Check if the dataframes are empty before writing
+    if amendments_df is not None:
+        filename = (
+            f"{survey_year}_freezing_amendments_to_review_{tdate}_v{run_id}.csv"
+        )
+        write_csv(os.path.join(freezing_changes_to_review_path, filename), amendments_df)
+
+    if additions_df is not None:
+        filename = (
+            f"{survey_year}_freezing_additions_to_review_{tdate}_v{run_id}.csv"
+        )
+        write_csv(os.path.join(freezing_changes_to_review_path, filename), additions_df)
+
+    return True
```

Comment:
should this return False if no files were written? (e.g., there are no additions or amendments)